### PR TITLE
cmd: make update config conservative, only update listed parameter

### LINKED ISSUE #1545: Unexpected behavior for cdc cli changefeed update
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   1. create a changefeed with `sort-engine=unified`
   2. update changefeed config via `cdc cli changefeed update -c xxx --sink-uri="blackhole://"

2. What did you expect to see?

   Only `sink-uri` is updated

3. What did you see instead?

   `sort-engine` is set to `memory`

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        v4.0.11
        ```

    - TiCDC version (execute `cdc version`):

        ```
        v4.0.11
        ```

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix #1545

### What is changed and how it works?

Change the behavior of `cdc cli changefeed update`, will only update parameter that provided in command.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- cli changefeed update will update parameter provided in command only.

### CODE REVIEW

**Comment by kennytm** [comment]
what the heck, bot 😓.

**Comment by amyangfei** [comment]
> what the heck, bot 😓.

Make me confused too. What's the rule that tichi-bot chooses review candidates? @hi-rustin

**Comment by hi-rustin** [comment]
> > what the heck, bot 😓.
> 
> Make me confused too. What's the rule that tichi-bot chooses review candidates? @hi-rustin



> > what the heck, bot 😓.
> 
> Make me confused too. What's the rule that tichi-bot chooses review candidates? @hi-rustin

See: https://book.prow.tidb.io/#/en/plugins/blunderbuss?id=design
kennytm is a member of migrate SIG and may be picked because it is weighted random. If you don't think you want this, you need to ignore kennytm in the configuration, which is not in the current configuration. 

https://github.com/ti-community-infra/configs/blob/49e8130252ba166e5de5f880b1fc7a6f9c61d361/prow/config/external_plugins_config.yaml#L727

**Comment by amyangfei** [comment]
PTAL @leoppro @liuzix

**Comment by kennytm** [comment]
@hi-rustin is it possible to make that an "include" list instead of an "exclude" list?

**Comment by hi-rustin** [comment]
> @hi-rustin is it possible to make that an "include" list instead of an "exclude" list?

For some large repos, this may be more difficult to list everyone.

File: cdc/model/changefeed_test.go [hunk 1]
```diff
@@ -190,6 +190,30 @@ func (s *configSuite) TestVerifyAndFix(c *check.C) {
 	c.Assert(marshalConfig1, check.Equals, marshalConfig2)
 }
 
+func (s *configSuite) TestChangeFeedInfoClone(c *check.C) {
+	defer testleak.AfterTest(c)()
+	info := &ChangeFeedInfo{
+		SinkURI: "blackhole://",
+		Opts:    map[string]string{},
```
Comment (lance6716) [inline cdc/model/changefeed_test.go:197]: i remember a empty map will become nil in some case. will it break something?

Comment (amyangfei) [reply cdc/model/changefeed_test.go:197]: Not the same thing, `map[string]string{}` or `new(map[string]string)` has initialized the map, it is not nil anymore

**Review by leoppro — APPROVED** [review]

File: cdc/model/changefeed_test.go (hunk 1 above)
Comment (lance6716) [reply cdc/model/changefeed_test.go:197]: I checked that only for Unmarshal of `protoc-gen-gogo`, empty map will become nil 😂

**Pull request merged**